genie-biogem/src/fortran/biogem_data.f90

```diff
@@ -971,54 +971,93 @@ SUBROUTINE sub_data_update_tracerrelationships()
     ! -------------------------------------------------------- !
     ! UPDATE REDFIELD RELATIONSHIPS
     ! -------------------------------------------------------- !
-    ! if NO3 is employed;
-    ! calculate alkalnity corrections associated with the formation and destruction of organic matter from NO3
-    ! otherwise, convert PO4 units to NO3 via the P:N Redfield ratio and then calculate the ALK correction from NO3
+    ! N
+    ! NOTE: leave alone the conv_ocn_sed (organic matter formation) connections
+    !       (e.g. if NH4 selected -- still assume default assimilation (and uptake) of NO3 to form PON)
+    if (ocn_select(io_NH4)) then
+       conv_sed_ocn(io_NO3,is_PON) = 0.0
+       conv_sed_ocn(io_NH4,is_PON) = 1.0
+    elseif (ocn_select(io_NO3)) then
+       conv_sed_ocn(io_NO3,is_PON) = 1.0
+       conv_sed_ocn(io_NH4,is_PON) = 0.0
+    else
+       conv_sed_ocn(io_NO3,is_PON) = 0.0
+       conv_sed_ocn(io_NH4,is_PON) = 0.0
+    end if
+    ! ALK
+    ! if NO3 is employed: calculate alkalnity corrections associated with the formation and destruction of organic matter from NO3
+    ! otherwise: convert PO4 units to NO3 via the P:N Redfield ratio and then calculate the ALK correction from NO3
     ! NOTE: ensure that both corrections are mutually exclusive (i.e., make sure that there can be no double ALK correction)
-    ! NOTE: catch incidence of par_bio_red_PON_ALK set to 0.0
-    if (abs(par_bio_red_PON_ALK) > const_real_nullsmall) then
-       if (ocn_select(io_NO3)) then
-          conv_sed_ocn(io_ALK,is_PON) = par_bio_red_PON_ALK
-          conv_ocn_sed(is_PON,io_ALK) = 1.0/conv_sed_ocn(io_ALK,is_PON)
-          conv_sed_ocn(io_ALK,is_POP) = 0.0
-          conv_ocn_sed(is_POP,io_ALK) = 0.0
-       else
-          conv_sed_ocn(io_ALK,is_PON) = 0.0
-          conv_ocn_sed(is_PON,io_ALK) = 0.0
+    ! NOTE: catch any incidence of Redfield ratios (par_bio_red_xxx) set to 0.0
+    ! NOTE: some of the zero values here are already the (hard-coded) defaults ... they are repeated here to be completely explicit
+    if (ocn_select(io_NH4)) then
+       conv_sed_ocn(io_ALK,is_PON) = conv_sed_ocn(io_NH4,is_PON)
+       conv_ocn_sed(is_PON,io_ALK) = 1.0/conv_sed_ocn(io_ALK,is_PON)
+       conv_sed_ocn(io_ALK,is_POP) = 0.0
+       conv_ocn_sed(is_POP,io_ALK) = 0.0
+       conv_sed_ocn(io_ALK,is_POC) = 0.0
+       conv_ocn_sed(is_POC,io_ALK) = 0.0
+    elseif (ocn_select(io_NO3)) then
+       conv_sed_ocn(io_ALK,is_PON) = -conv_sed_ocn(io_NO3,is_PON)
```

Comment:
The correction for Alk with NH4 looks good to me. I'm not sure about the correction when only NO3 is selected. My understanding is that Alk = HCO3- + 2 CO32- + B(OH)4- + OH- - H+ + HPO42- + 2PO43- + H3SiO4- + NH3 +HS- ...

When we remineralise PON into NO3, we assume that nitrification is instantaneous: PON->NH3->NO3-. Looking at the full equation for nitrification we get:
NH3 + O2 + H2O -> NO3- + 5H+ + 4e-

So do we not need to remove some Alk due to 5H+ production?